app/queries/new_accounts_query.rb

```diff
@@ -60,22 +60,22 @@ def granularity_to_date_format(granularity)
     end
   end
 
-  def mysql_subquery(range, date_format, timezone: Time.zone.tzinfo.name)
+  def mysql_subquery(range, date_format, timezone: Time.zone)
     account.buyer_accounts
-        .where.has { sift(:date, sift(:in_timezone, created_at, name: timezone)).in(range) }
-        .grouping { sift(:date_format, sift(:in_timezone, created_at, name: timezone), date_format).to_sql }
+        .where.has { sift(:date, sift(:in_timezone, created_at, timezone)).in(range) }
+        .grouping { sift(:date_format, sift(:in_timezone, created_at, timezone), date_format).to_sql }
```

Comment:
would be nice to explain this as well :)

Reply:
The error observed was:
```
NoMethodError (undefined method `tzinfo' for {:name=>"Etc/UTC"}:Hash
```

The issue was basically that a mix of positional and keyword arguments was used, and after switching to Ruby 3 it stopped working fine.

This was the previous definition:
```
sifter :in_timezone do |column, zone = Time.zone, name: zone.tzinfo.name, offset: zone.formatted_offset|
```

and the invocation (`created_at` corresponds to `column`):
```
sift(:in_timezone, created_at, name: timezone)
```
I think this is actually extremely confusing - why are we expecting the `name` to be taken from the passed argument, but `zone` auto-assigned, and `offset derived from `zone`?

Now it's much simpler, less confusing and kinda makes sense (and works :D)

Reply:
This refactor broke oracle and postres queries, so I reverted the change and refactored it in a different way :)